src/lib/utils.js

```diff
@@ -123,18 +123,27 @@ function prepareTemplates(defaultTemplates, templates) {
   }, {templates: {}, useCustomCompileOptions: {}});
 }
 
-function getRefinement(attributeName, name, resultsFacets) {
-  let res = {attributeName, name};
-  let facet = find(resultsFacets, (_facet) => { return _facet.name === attributeName; });
-  if (facet !== undefined) {
-    const count = get(facet, 'data.' + name);
-    const exhaustive = get(facet, 'exhaustive');
-    if (count !== undefined) {
-      res.count = count;
-    }
-    if (exhaustive !== undefined) {
-      res.exhaustive = exhaustive;
+function getRefinement(state, type, attributeName, name, resultsFacets) {
+  let res = {type, attributeName, name};
+  let facet = find(resultsFacets, {name: attributeName});
+  let count;
+  if (type === 'hierarchical') {
+    let facetDeclaration = state.getHierarchicalFacetByName(attributeName);
+    let splitted = name.split(facetDeclaration.separator);
+    res.name = splitted[splitted.length - 1];
```

Comment:
https://github.com/algolia/instantsearch.js/blob/fix/getRefinements-hierarchical/src/lib/utils.js#L134-L137

Basically, for
```js
{
  name: 'myHierarchicalFacet',
  attributes: ['myHierarchicalFacet.lvl0', 'myHierarchicalFacet.lvl1', 'myHierarchicalFacet.lvl2'],
  separator: ' > '
}
```
and a refinement like this `Cat > SubCat`;
the hierarchical data returned by the helper looks like this:
```js
{
  name: 'myHierarchicalFacet',
  data: [
    {
      name: 'Cat',
      count: 2,
      data: [
        { name: 'Subcat', count: 1 }
      ]
    }
  ]
}
```
This is really different from another type of facet where it is like this:
```js
{
  name: 'myFacet',
  data: {
    Cat: 2
  }
}
```

Hence the need for a special handling.